commit a0737c63f10dd6c033dbb5653120da7a959cae6a
Author: Kai Uwe Broulik <kde@privat.broulik.de>
Date:   Sat Dec 7 18:34:09 2024 +0100

    NotifyByAndroid: Set CONTEXT_NOT_EXPORTED
    
    It causes a SecurityException on Android 15.

diff --git a/src/android/org/kde/knotifications/NotifyByAndroid.java b/src/android/org/kde/knotifications/NotifyByAndroid.java
index c51eb342..2d5697f9 100644
--- a/src/android/org/kde/knotifications/NotifyByAndroid.java
+++ b/src/android/org/kde/knotifications/NotifyByAndroid.java
@@ -69,7 +69,12 @@ public class NotifyByAndroid extends BroadcastReceiver
         filter.addAction(m_ctx.getPackageName() + NOTIFICATION_DELETED);
         filter.addAction(m_ctx.getPackageName() + NOTIFICATION_OPENED);
         filter.addAction(m_ctx.getPackageName() + NOTIFICATION_REPLIED);
-        m_ctx.registerReceiver(this, filter);
+
+        if (Build.VERSION.SDK_INT >= 33) {
+            m_ctx.registerReceiver(this, filter, Context.RECEIVER_NOT_EXPORTED);
+        } else {
+            m_ctx.registerReceiver(this, filter);
+        }
     }
 
     public void notify(KNotification notification)
